test/jasmine/tests/bar_test.js

```diff
@@ -612,6 +702,168 @@ describe('A bar plot', function() {
 
     afterEach(destroyGraphDiv);
 
+    function getAllTraceNodes(node) {
+        return node.querySelectorAll('g.points');
+    }
+
+    function getAllBarNodes(node) {
+        return node.querySelectorAll('g.point');
+    }
+
+    function assertTextIsInsidePath(textNode, pathNode) {
+        var textBB = textNode.getBoundingClientRect(),
+            pathBB = pathNode.getBoundingClientRect();
+
+        expect(pathBB.left).not.toBeGreaterThan(textBB.left);
+        expect(textBB.right).not.toBeGreaterThan(pathBB.right);
+        expect(pathBB.top).not.toBeGreaterThan(textBB.top);
+        expect(textBB.bottom).not.toBeGreaterThan(pathBB.bottom);
```

Comment:
Since I've used `Drawing.bBox(node)` to compute the location of bar texts and `Drawing.bBox(node)` uses `getBoundingClientRect` this shouldn't be an issue.

I noticed, however, in certain cases, a rounding error that would cause the bar text to be negligibly larger than computed. For that reason, and also because it looks neater, I added some padding to bar texts. See [here](https://github.com/n-riesco/plotly.js/blob/7d7017c5b97b7204a3e74ef44d3d03de720d2be7/src/traces/bar/plot.js#L279-L281) and [here](https://github.com/n-riesco/plotly.js/blob/7d7017c5b97b7204a3e74ef44d3d03de720d2be7/src/traces/bar/plot.js#L298-L323).